python/pyiceberg/utils/datetime.py

```diff
@@ -47,11 +47,16 @@ def micros_to_time(micros: int) -> time:
     return time(hour=hours, minute=minutes, second=seconds, microsecond=microseconds)
 
 
-def date_to_days(date_str: str) -> int:
+def date_str_to_days(date_str: str) -> int:
```

Comment:
Does this need to be renamed? Why not support both with the same name to avoid a breaking change?

Reply:
My thought process was that this API is still pre 1.0 and the existing name was misleading, so better to improve it now.  I have no preference either way though, so happy to adjust.  I'm not sure how to support both with the same name though?

Reply:
Yeah, it probably isn't used anywhere. You're right: better to clean it up.